lib/improver/blending/weighted_blend.py

```diff
@@ -454,165 +454,216 @@ def process(self, cube, weights=None):
                         one value. Returns the original cube in this case.
 
         """
+
+        self._check_input_cube_is_iris_cube(cube)
+
+        # if the coordinate being blended is forecast_reference_time.
+        if self.coord == "forecast_reference_time":
+            self._check_all_time_points_equal(cube)
+
+        if weights is not None:
+            self._check_weights_have_coord_shape(cube, weights)
+
+        # only do blending if coordinate is a dimension coordinate
+        coord_dim = cube.coord_dims(self.coord)
+        if coord_dim:
+            result = self._perform_blending(cube, weights)
+        else:
+            # scalar coordinate
+            msg = ('Trying to blend across a scalar coordinate with only one'
+                   ' value. Returning original cube')
+            warnings.warn(msg)
+            result = cube
+
+        # adjust values of collapsed coordinates if required
+        if self.coord_adjust is not None:
+            result = self._adjust_coord(result, cube, coord_dim)
+
+        return result
+
+    @staticmethod
+    def _check_input_cube_is_iris_cube(cube):
         if not isinstance(cube, iris.cube.Cube):
             msg = ('The first argument must be an instance of '
                    'iris.cube.Cube but is'
                    ' {0:s}.'.format(type(cube)))
             raise TypeError(msg)
```

Comment:
This is obviously unaltered functionality - but is the same check done elsewhere? Needs to be somewhere else rather than a staticmethod? Could be captured in a follow on issue if necessary.